signature/sign.go

```diff
@@ -113,16 +120,28 @@ func Sign(_ context.Context, key jwk.Key, sf SignedFielder, opts ...Option) (*pi
 		return nil, err
 	}
 
-	if options.logger != nil {
+	switch key := key.(type) {
+	case jwk.Key:
 		pk, err := key.PublicKey()
```

Comment:
This method isn't implemented in the above interface

Reply:
Within this `case`, `key` has the type `jwk.Key`, since it was redeclared within the `switch` line (`key := key.(type)`). This is probably the most magical of all Go syntax.